### PR TITLE
Add p4info files as reference outputs

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

File: frontends/p4/typeChecking/typeChecker.cpp [hunk 1]
```diff
@@ -118,8 +118,8 @@ const IR::Type* TypeInference::cloneWithFreshTypeVariables(const IR::IMayBeGener
 }
 
 TypeInference::TypeInference(ReferenceMap* refMap, TypeMap* typeMap, bool readOnly) :
-        refMap(refMap), typeMap(typeMap), readOnly(readOnly),
-        initialNode(nullptr) {
+        refMap(refMap), typeMap(typeMap),
+        initialNode(nullptr), readOnly(readOnly) {
```
Comment (mbudiu-vmw) [inline frontends/p4/typeChecking/typeChecker.cpp]: this is cleaning up a warning introduced in a recent commit by @hanw

**Comment by hesingh** [comment]
@mbudiu-vmw Can you also please change this code in typeChecker.cpp for checkEmitType() to change the 'continue;' to a BUG like you told me in our code in the same file in a  review.  thanks.

```
if (type->is<IR::Type_Struct>()) {
        for (auto f : type->to<IR::Type_Struct>()->fields) {
            auto ftype = typeMap->getType(f);
            if (ftype == nullptr)
                continue;
            checkEmitType(emit, ftype);
        }
        return;
    }
```

**Comment by mbudiu-vmw** [comment]
@hesingh : your request does not seem to have anything to do with the issue that we are attempting to solve here.

**Comment by hesingh** [comment]
> @hesingh : your request does not seem to have anything to do with the issue that we are attempting to solve here.

I thought since you were making another change in the same .cpp file, one could make another change.  However, it's fine to keep this change out of this PR and use another PR to make the change.  thanks.

**Comment by antoninbas** [comment]
I just realized that the p4Runtime serializer doesn't really sort the objects (e.g. tables) in any specific order. But it seems that Travis passed, so hopefully the order is relatively stable (because the IR is traversed deterministically?).

**Comment by mbudiu-vmw** [comment]
The IR traversal is deterministic. We tried to make the entire compiler deterministic.

**Comment by antoninbas** [comment]
I'm just hoping that most changes to the compiler (e.g. new transform visitor) will not cause all the P4Info files to be changed. If I add a transform pass, it creates new instances of the IR nodes, which may impact the ordering in the IR data structures (e.g. IndexedVector), which may impact P4Info generation. How are objects sorted in the IR data structures?

**Comment by antoninbas** [comment]
NVM, IndexedVector was a bad example since it is ordered based on the name...

**Comment by jnfoster** [comment]
Nice.

**Pull request merged**